demo/body3d_two_stage_video_demo.py

```diff
@@ -167,34 +207,75 @@ def main():
         args.pose_detector_checkpoint,
         device=args.device.lower())
 
-    assert pose_det_model.cfg.model.type == 'TopDown', 'Only "TopDown"' \
+    assert isinstance(pose_det_model, TopDown), 'Only "TopDown"' \
         'model is supported for the 1st stage (2D pose detection)'
 
+    # some hand-crafted setting to save memory for specific models
+    if args.save_memory:
+        # If use 'PoseWarper' detector, set concat_tensors to False
+        if pose_det_model.__class__.__name__ == 'PoseWarper':
+            pose_det_model.concat_tensors = False
+
+    # frame index offsets for inference, used in multi-frame inference setting
+    if args.use_multi_frames:
+        assert 'frame_indices_test' in pose_det_model.cfg.data.test.data_cfg
+        indices = pose_det_model.cfg.data.test.data_cfg['frame_indices_test']
+
     pose_det_dataset = pose_det_model.cfg.data['test']['type']
+    # get datasetinfo
+    dataset_info = pose_det_model.cfg.data['test'].get('dataset_info', None)
+    if dataset_info is None:
+        warnings.warn(
+            'Please set `dataset_info` in the config.'
+            'Check https://github.com/open-mmlab/mmpose/pull/663 for details.',
+            DeprecationWarning)
+    else:
+        dataset_info = DatasetInfo(dataset_info)
 
     pose_det_results_list = []
     next_id = 0
     pose_det_results = []
-    for frame in mmcv.track_iter_progress(video):
+
+    print('Running 2D pose detection inference...')
+    for frame_id, cur_frame in enumerate(mmcv.track_iter_progress(video)):
         pose_det_results_last = pose_det_results
 
         # test a single image, the resulting box is (x1, y1, x2, y2)
-        mmdet_results = inference_detector(person_det_model, frame)
+        mmdet_results = inference_detector(person_det_model, cur_frame)
 
         # keep the person class bounding boxes.
         person_det_results = process_mmdet_results(mmdet_results,
                                                    args.det_cat_id)
 
-        # make person results for single image
+        if args.use_multi_frames:
```

Comment:
Is it possible to extract the multi-frame collection codes into a function to simplify the main function?

Reply:
I will move them to a separate function.